[DDP] log bucket sizes

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #62232

Logs the bucket sizes in DDP logging so that we know which workflow ran with what bucket size config. Will be used to verify how changing bucket sizes in DDP affects perf.

Based on the test, we can see inconsistency where the "first" bucket size actually is (last before rebuild buckets, first after).

Differential Revision: [D29922299](https://our.internmc.facebook.com/intern/diff/D29922299/)

rohan-varma committed:
[DDP] log bucket sizes

Logs the bucket sizes in DDP logging so that we know which workflow ran with what bucket size config. Will be used to verify how changing bucket sizes in DDP affects perf.

Based on the test, we can see inconsistency where the "first" bucket size actually is (last before rebuild buckets, first after).

Differential Revision: [D29922299](https://our.internmc.facebook.com/intern/diff/D29922299/)

[ghstack-poisoned]

mrshenli commented on torch/testing/_internal/distributed/distributed_test.py:
```diff
@@ -7292,6 +7296,56 @@ def get_loss(model_output):
+                def __init__(self):
+                    super().__init__()
+                    self.model = nn.Sequential(nn.Linear(2, 4000, bias=False),
+                        *[
```
Shall we fix lint errors before landing?

rohan-varma replied:
yep, looks like there's a test failure as well

rohan-varma committed:
Update on "[DDP] log bucket sizes"


Logs the bucket sizes in DDP logging so that we know which workflow ran with what bucket size config. Will be used to verify how changing bucket sizes in DDP affects perf.

Based on the test, we can see inconsistency where the "first" bucket size actually is (last before rebuild buckets, first after).

Differential Revision: [D29922299](https://our.internmc.facebook.com/intern/diff/D29922299/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[DDP] log bucket sizes"


Logs the bucket sizes in DDP logging so that we know which workflow ran with what bucket size config. Will be used to verify how changing bucket sizes in DDP affects perf.

Based on the test, we can see inconsistency where the "first" bucket size actually is (last before rebuild buckets, first after).

Differential Revision: [D29922299](https://our.internmc.facebook.com/intern/diff/D29922299/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[DDP] log bucket sizes"


Logs the bucket sizes in DDP logging so that we know which workflow ran with what bucket size config. Will be used to verify how changing bucket sizes in DDP affects perf.

Based on the test, we can see inconsistency where the "first" bucket size actually is (last before rebuild buckets, first after).

Differential Revision: [D29922299](https://our.internmc.facebook.com/intern/diff/D29922299/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[DDP] log bucket sizes"


Logs the bucket sizes in DDP logging so that we know which workflow ran with what bucket size config. Will be used to verify how changing bucket sizes in DDP affects perf.

Based on the test, we can see inconsistency where the "first" bucket size actually is (last before rebuild buckets, first after).

Differential Revision: [D29922299](https://our.internmc.facebook.com/intern/diff/D29922299/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[DDP] log bucket sizes"


Logs the bucket sizes in DDP logging so that we know which workflow ran with what bucket size config. Will be used to verify how changing bucket sizes in DDP affects perf.

Based on the test, we can see inconsistency where the "first" bucket size actually is (last before rebuild buckets, first after).

Differential Revision: [D29922299](https://our.internmc.facebook.com/intern/diff/D29922299/)

[ghstack-poisoned]
